iam/cloud-client/snippets/set_policy.py

```diff
@@ -19,24 +19,36 @@
 from google.iam.v1 import iam_policy_pb2, policy_pb2
 
 
-def set_policy(project_id: str, policy: policy_pb2.Policy) -> policy_pb2.Policy:
+def set_project_policy(project_id: str, policy: policy_pb2.Policy, merge: bool = True) -> policy_pb2.Policy:
     """
     Set policy for project. Pay attention that previous state will be completely rewritten.
     If you want to update only part of the policy follow the approach read->modify->write.
     For more details about policies check out https://cloud.google.com/iam/docs/policies
 
     project_id: ID or number of the Google Cloud project you want to use.
     policy: Policy which has to be set.
+    merge: The strategy to be used forming the request
```

Comment:
CopyFrom is clearing both mutable and immutable fields, when MergeFrom is replacing only immutable fields and extending mutable (https://googleapis.dev/python/protobuf/latest/google/protobuf/message.html#google.protobuf.message.Message.CopyFrom).

I added this attribute to handle few scenarios: 
- Since tests are running in parallel, it is important to not rewrite policies from scratch, but extend them to avoid collisions on assertions (MergeFrom used)
- After finishing all the tests it is needed to "hardly" reset to the previous state (CopyFrom used)

Reply:
OK, makes sense, please add that first paragraph to the docstring, so users understand it as well :)